odl/tomo/geometry/conebeam.py

```diff
@@ -544,12 +568,21 @@ def __getitem__(self, indices):
         apart = part.byaxis[0]
         dpart = part.byaxis[1]
 
-        return FanFlatGeometry(apart, dpart,
-                               src_radius=self.src_radius,
-                               det_radius=self.det_radius,
-                               src_to_det_init=self.src_to_det_init,
-                               det_axis_init=self._det_axis_init_arg,
-                               translation=self.translation)
+        return FanGeometry(apart, dpart,
+                           src_radius=self.src_radius,
+                           det_radius=self.det_radius,
+                           src_to_det_init=self.src_to_det_init,
+                           det_axis_init=self._det_axis_init_arg,
+                           translation=self.translation)
+
+
+class FanFlatGeometry(FanGeometry):
+
+    """Fan beam (2d cone beam) geometry with flat 1d detector.
+
+    FanGeometry with the default det_curve_radius = None """
```

Comment:
I copied the general description of a class, however, most of the documentation is under the functions, for instance,  under the __init__(). Should I just copy it as well? I don't know how the documentation  process work, to be honest

Reply:
Oh dear. I remember we had a discussion some time ago whether we should put the documentation of a class (constructor parameters, notes, etc.) in the class docstring or in the docstring of `__init__`. I was arguing for `__init__` back then, but this case is actually a good reason against it.

So yes, please do copy over the `__init__` docstring as well (without the first line).

As for the documentation process, the `sphinx` tool will collect the docstrings for classes and their methods and generate the HTML accordingly. The important detail here is that it will not add or somehow merge with the documentation of parent classes, but only look at the class itself.